src/XrdHttpTpc/XrdHttpTpcUtils.hh

```diff
@@ -32,6 +32,19 @@ namespace TPC {
 
 class XrdHttpTpcUtils {
 public:
+  struct PrepareOpenURLParams {
+    PrepareOpenURLParams(const std::string & reqResourceParm,
+                         std::map<std::string,std::string> & reqHeadersParm,
+                         const std::map<std::string,std::string> & hdr2cgimapParm,
+                         const std::map<std::string,std::string> & reprDigestParm):
+      reqResource(reqResourceParm),reqHeaders(reqHeadersParm),hdr2cgimap(hdr2cgimapParm),reprDigest(reprDigestParm){}
+
+    const std::string & reqResource;
+    std::map<std::string,std::string> & reqHeaders;
+    const std::map<std::string,std::string> & hdr2cgimap;
+    const std::map<std::string,std::string> & reprDigest;
+  };
```

Comment:
If this struct is part of the class, then `prepareOpenURL` should not be a `static` member. I'd either put the struct outside the class and only in the `.cc` file, and use a free `static` function there (to hide it from the public interface), or I'd make the struct and function private (or protected) within `XrdHttpUtils`.